qubessalt/__init__.py

```diff
@@ -234,7 +234,26 @@ def salt_call(self, command='state.highstate', return_output=False):
         return exit_code, return_data
 
 
-def run_one(vmname, command, show_output, force_color):
+def load_opts():
+    opts = salt.config.minion_config('/etc/salt/minion')
+    opts['file_client'] = 'local'
+    return opts
+
+
+def has_config(vm):
+    opts = load_opts()
+    opts['id'] = vm
+    caller = salt.client.Caller(mopts=opts)
+    top = caller.cmd('state.show_top', queue=False, concurrent=True)
+    return bool(top)
+
+
+def run_one(vmname, command, show_output, force_color, skip_top_check):
+    try:
+        if not skip_top_check and 'state.highstate' in command and not has_config(vmname):
+            return vmname, 0, "SKIP (nothing to do)"
+    except Exception as err:  # pylint: disable=broad-except
+        return vmname, 1, f"ERROR (exception {err})"
```

Comment:
This seems to be a bit too broad to me.  I would only wrap `has_config` in the try-except block.

Reply:
Itched me a bit as well, although in this multiprocessing case any exception thrown would just be silenced anyways afaik.

Happy to adapt though, the only exception I can imagine in the other parts would be if `command` is of the wrong type, which is passed in by the same module, so shouldn't happen.